documentation/modules/operators/con-topic-operator-store.adoc

```diff
@@ -0,0 +1,115 @@
+
+== Internal topic store topics
+
+Internal topics support the handling of topic metadata.
+
+`__strimzi_store_topic`:: Input topic for storing the topic metadata
+`__strimzi-topic-operator-kstreams-topic-store-changelog`:: Retains a log of compacted topic store values
+
+== Migrating topic metadata from ZooKeeper
+
+In previous releases of Strimzi, topic metadata was stored in ZooKeeper.
+The new process removes this requirement, bringing the metadata into the Kafka cluster, and under the control of the Topic Operator.
+
+When upgrading to Strimzi {ProductVersion}, the transition to Topic Operator control of the topic store is seamless.
+Metadata is found and migrated from ZooKeeper, and the old store is cleansed.
+
+== Downgrading to a Strimzi version that uses ZooKeeper to store topic metadata
+
+If you are reverting back to a version of Strimzi that uses ZooKeeper for the storage of topic metadata,
+you downgrade your Cluster Operator to the previous version,
+then downgrade Kafka brokers and client applications to the previous Kafka version as standard.
+
+Additionally, you must delete the topics created for the topic store using an `exec` command, specifying the bootstrap address of the Kafka cluster and the name of a broker pod.
+For example, where `my-cluster-kafka-0` is the name of the broker pod:
+
+[source,shell]
+----
+kubectl exec -ti my-cluster-kafka-0 -- ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --topic __strimzi-topic-operator-kstreams-topic-store-changelog --delete && ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --topic __strimzi_store_topic --delete
+----
+
+The Topic Operator will reconstruct the Zookeeper topic metadata from the state of the topics in Kafka.
+
+NOTE: When scaling a Kafka cluster, the topic store topics must also be deleted so that partitions can be reassigned.
+
+== Scaling and topics
+
+The recommended configuration for topics is to have a topic replication factor of 3, so that there are 2 in-sync replicas on other brokers.
+If a broker is unavailable, at least one other broker is in-sync.
+
+[source,shell,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaTopicApiVersion}
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 1
+  replicas: 3 <1>
+  config:
+    min.insync.replicas=2 <2>
+  #...
+----
+<1> The number of replica topics. Currently, this cannot be changed in the `KafkaTopic` resource, but it can be changed using the `kafka-reassign-partitions.sh` tool.
+<2> The minimum number of replica partitions that a message must be successfully written to, or an exception is raised.
+
+NOTE: In-sync replicas are used in conjunction with the `acks` configuration for a producer application.
+The `acks` configuration determines the number of follower partitions a message must be replicated to before the message is acknowledged as successfully received.
```

Comment:
This is true. But what does it mean for the user? We should either add how they can configure it for the TO. Or we should just say that the TO always runs with `acks=all`.

Reply:
Not sure if I'm capturing what you think we need in this update:

> NOTE: In-sync replicas are used in conjunction with the `acks` configuration for a producer application.
> The `acks` configuration determines the number of follower partitions a message must be replicated to before the message is acknowledged as successfully received. The Topic Operator runs with `acks=all` by default, whereby messages must be acknowledged by all in-sync replicas.

Reply:
`runs with acks=all by default` ... I think it is not possible to change it. So I would leave out the `by default` maybe?

Reply:
Okay.